### DISCUSSION TITLE
Fullstack.Net - Dúvida sobre DateOnly

### DISCUSSION THREAD

**Discussion by MelloRenato** [balta-io/forum#660] [Tira Dúvidas] [answered]
Prezados,

Vejo que mesmo não sendo relevante a hora em muitos dos campos de data, os desenvolvedores utilizam o tipo DateTime, podemos utilizar os campos CreatedAt e o PaidOrReceivedAt.

Por que não utilizar o tipo DateOnly? 
Quais suas desvantagens/vantagens em relação ao DateTime?

Grato,
Renato Mello

**Comment by henriqueweege** [comment, marked as answer]
Fala, @MelloRenato ! Tudo certinho?
De acordo com [esse artigo da Microsoft](https://learn.microsoft.com/en-us/dotnet/standard/datetime/choosing-between-datetime), usar o DateOnly tem alguns benefícios, como serializar menos dados e não ter problema de offset por timezone. Além disso, segundo a doc, ainda é um tipo que "conversa melhor" com o tipo "date" dos bancos de dados relacionais, por não conter hora. 
Sobre os exemplos que você deu: dependendo do contexto do domínio a hora faz diferença sim. Por exemplo ao registrar um pagamento de uma parcela de empréstimo é importante ter a hora do mesmo, pois dependendo de quando foi realizado vai ter acruo de juros ou não.
Enfim, cada caso é um caso e vale a pena avaliar individualmente qual caso se encaixa melhor cada tipo.

Valeeeu!

**Reply by MelloRenato** [reply]
Obrigado @henriqueweege .
